controllers/operands/wasp.go

```diff
@@ -0,0 +1,225 @@
+			crType: "DaemonSet",
+			hooks:  &waspHooks{},
+		},
+		getCRWithName: func(hc *hcov1beta1.HyperConverged) client.Object {
+			return NewWaspWithNameOnly(hc)
+		},
+		shouldDeploy: func(hc *hcov1beta1.HyperConverged) bool {
+			return hc.Spec.FeatureGates.EnableHigherDensityWithSwap != nil &&
+				*hc.Spec.FeatureGates.EnableHigherDensityWithSwap
+		},
+	}
+}
+
+type waspHooks struct {
+	cache *appsv1.DaemonSet
+}
+
+func (h *waspHooks) getFullCr(hc *hcov1beta1.HyperConverged) (client.Object, error) {
+	if h.cache == nil {
+		h.cache = NewWasp(hc)
+	}
+
+	return h.cache, nil
+}
+
+func (*waspHooks) getEmptyCr() client.Object { return &appsv1.DaemonSet{} }
+
+func (*waspHooks) updateCr(
+	req *common.HcoRequest,
+	Client client.Client,
+	exists runtime.Object,
+	required runtime.Object) (bool, bool, error) {
+	wasp, ok1 := required.(*appsv1.DaemonSet)
+	found, ok2 := exists.(*appsv1.DaemonSet)
+
+	if !ok1 || !ok2 {
+		return false, false, errors.New("can't convert to DaemonSet")
+	}
+
+	if !hasCorrectDaemonSetFields(found, wasp) {
+		overwritten := false
+		if req.HCOTriggered {
+			req.Logger.Info("Updating existing wasp DaemonSet Spec to new opinionated values")
+		} else {
+			req.Logger.Info("Reconciling an externally updated wasp DaemonSet Spec to its opinionated values")
+			overwritten = true
+		}
+		hcoutil.MergeLabels(&wasp.ObjectMeta, &found.ObjectMeta)
+		wasp.Spec.DeepCopyInto(&found.Spec)
+		err := Client.Update(req.Ctx, found)
+		if err != nil {
+			return false, false, err
+		}
+		return true, overwritten, nil
+	}
+
+	return false, false, nil
+}
+
+func (*waspHooks) justBeforeComplete(_ *common.HcoRequest) { /* no implementation */ }
+
+func (*waspHooks) getConditions(cr runtime.Object) []metav1.Condition {
+	return dsConditionsToK8s(cr.(*appsv1.DaemonSet).Status.Conditions)
+}
+
+func NewWasp(hc *hcov1beta1.HyperConverged) *appsv1.DaemonSet {
+	waspImage, _ := os.LookupEnv(hcoutil.WaspImageEnvV)
+	wasp := NewWaspWithNameOnly(hc)
+	wasp.Spec = appsv1.DaemonSetSpec{
+		Selector: &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				"name": "wasp",
+			},
+		},
+		UpdateStrategy: appsv1.DaemonSetUpdateStrategy{
+			Type: appsv1.RollingUpdateDaemonSetStrategyType,
+			RollingUpdate: &appsv1.RollingUpdateDaemonSet{
+				MaxUnavailable: ptr.To(intstr.FromString("10%")),
+				MaxSurge:       ptr.To(intstr.FromString("0")),
+			},
+		},
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{
+					"name": "wasp",
+				},
+			},
+			Spec: corev1.PodSpec{
+				RestartPolicy:                 corev1.RestartPolicyAlways,
+				ServiceAccountName:            waspServiceAccount,
+				HostPID:                       true,
+				HostUsers:                     ptr.To(true),
+				TerminationGracePeriodSeconds: ptr.To(int64(30)),
+				Containers: []corev1.Container{
+					{
+						Name:            "wasp-agent",
+						Image:           waspImage,
+						ImagePullPolicy: corev1.PullAlways,
```

Comment:
why not `IfNotPresent` as all other pods in kubevirt?
this is important for disconnected envs.